meinberlin/react/livequestions/QuestionModerator.jsx

```diff
@@ -1,190 +1,189 @@
-import React from 'react'
+import React, { useState, useEffect, useCallback } from 'react'
 import django from 'django'
 import LikeCard from './LikeCard'
 import { classNames } from '../contrib/helpers'
 
-export default class QuestionModerator extends React.Component {
-  constructor (props) {
-    super(props)
+const hiddenText = django.gettext('mark as hidden')
+const undoHiddenText = django.gettext('undo mark as hidden')
+const doneText = django.gettext('mark as done')
+const addLiveText = django.gettext('added to live list')
+const removeLiveText = django.gettext('remove from live list')
+const addBookmarkText = django.gettext('add bookmark')
+const removeBookmarkText = django.gettext('remove bookmark')
 
-    this.state = {
-      is_on_shortlist: this.props.is_on_shortlist,
-      is_live: this.props.is_live,
-      likes: this.props.likes.count,
-      session_like: this.props.likes.session_like,
-      is_hidden: this.props.is_hidden,
-      is_answered: this.props.is_answered
-    }
-  }
+const QuestionModerator = (props) => {
```

Comment:
Maybe destructure the props argument

Reply:
@vellip Destructuring to this:

```
const QuestionModerator = ({
  category,
  children,
  displayIsAnswered,
  displayIsHidden,
  displayIsLive,
  displayIsOnShortlist,
  handleLike,
  id,
  is_answered,
  is_hidden,
  is_live,
  is_on_shortlist,
  likes,
  removeFromList,
  togglePollingPaused,
  updateQuestion
}) => {
```

Causes eslint errors:

`error  Identifier 'is_answered' is not in camel case `

But changing to `isAnswered` then conflicts with the `isAnswered` local state naming:

`const [isAnswered, setIsAnswered] = useState(is_answered)`

Let's keep as `props` for now?

Reply:
```jsx
const QuestionModerator = ({
  category,
  children,
  displayIsAnswered,
  displayIsHidden,
  displayIsLive,
  displayIsOnShortlist,
  handleLike,
  id,
  **is_answered: _isAnswered,**
  is_hidden,
  is_live,
  is_on_shortlist,
  likes,
  removeFromList,
  togglePollingPaused,
  updateQuestion
}) => {
```

does that work?

Reply:
without the ** of course

Reply:
Oh nice, yeah that works 👍

https://github.com/liqd/a4-meinberlin/pull/5751/files#diff-11e1fc49d7ab8e826f53c0b167259c0e8f36bff2da36beac044a81d4531cd694R23-R26